Refactor: Review내용 반영 및 파일/폴더 변경, 변수/함수 이름 변경 등

wh5929 opened the pull request:
### 리팩토링
1. Running 폴더, 파일 구조 정리(버거 코드)
2. RunningView btnSelector, menuBtn 타입뷰 -> 서브뷰로 변경
3. hide() helper 이름 및 구조 변경
4. RunGuideTabView tabView, tabViewIndicator 서브뷰로 변경 및 이름 변경
5. RunGuideTabView extension 사용하여 뷰/함수/네임스페이스 선언
6. FullScreenCoverView -> RunDetailDescriptionView 로 이름 변경
7. RunDetailDescriptionView 내부 변수, 함수명 변경
8. bottomViewContents, closeButton 타입 뷰 -> 서브 뷰 로 변경
등등

wh5929 committed:
### 기능 추가
1. RunningGuideView 화면 구성
2. Custom Scroll 구현
3. 바로 시작 뷰(JustStartView)의 runGuideExperienceTabItem 탭 이벤트 발생시 해당 항목을 fullScreenCover를 이용해 상세화면 표시
4. Util에 Show+Extension.swift 파일 추가 -> Bool 프로퍼티를 이용해 View를 표시/숨기도록 하는 ViewModifier와 extension

### 내용 수정
1. RunGuideItem 프로퍼티 항목 추가 및 이름 변경
2. RunGuideViewModel 에서 추가된 프로퍼티 항목 기입 및 내용 추가 / 불필요한 코드 삭제
3. 필요없거나 저작권 있는 사진 삭제 / Asset 사진 이름 변경

wh5929 committed:
### Review 내용 반영
1. enum namespace 이용하여 리터럴 관리
2. 각종 뷰, UI / 로직 분리
3. Show+Extension 이름 변경 -> View+Extension / Show 구조체 이름 변경 -> Hide
4. 너무 긴 프로퍼티 이름 축소

wh5929 committed:
### 메서드명 수정
show() -> hide()

wh5929 committed:
### 리팩토링
메인 뷰에서 CloseButton() 정렬하기 위해 컨테이너뷰를 사용했으나, 더 좋은 방법을 찾게 되어 수정.

wh5929 committed:
### 리팩토링

1. 디렉터리 생성 및 파일 정리
2. Preset 값 enum namespace 로 정의
3. 불필요한 선언 제거
4. RunGuideTabView 서브뷰 추출 및 정리

wh5929 committed:
### 리팩토링
1. 폴더, 파일 구조 정리(버거 코드)
2. RunningView btnSelector, menuBtn 타입뷰 -> 서브뷰로 변경
3. hide() helper 이름 및 구조 변경
4. RunGuideTabView tabView, tabViewIndicator 서브뷰로 변경 및 이름 변경
5. RunGuideTabView extension 사용하여 뷰/함수/네임스페이스 선언
6. FullScreenCoverView -> RunDetailDescriptionView 로 이름 변경
7. RunDetailDescriptionView 내부 변수, 함수명 변경
8. bottomViewContents, closeButton 타입 뷰 -> 서브 뷰 로 변경
등등

hanjaesueng commented on GatheRunner/Presentaion/Running/TabItem/RunGuideExperienceTabItem.swift:
```diff
@@ -0,0 +1,96 @@
+}
+
+extension RunGuideExperienceTabItem {
+    private enum Size {
+        static let tabItemTextsVStackFrameWidthRatio = 0.45
+        static let tabItemTextsVStackFrameHeightRatio = 0.14
+        static let tabItemTextsHStackFrameWidthRatio = 0.82
+        static let tabItemTextsHStackFrameHeightRatio = 0.13
+        static let tabItemImagesFrameWidthRatio = 0.28
+        static let tabItemImagesFrameHeightRatio = 0.14
+    }
```
질문
----------------
enum은 private인데 , 변수들은 static인게 좀 의아합니다. 의도가 무엇일까요??

wh5929 replied:
> ## 질문
> enum은 private인데 , 변수들은 static인게 좀 의아합니다. 의도가 무엇일까요??

### 답변
enum을 namespace용도로만 사용하기 위해 변수들을 static 으로 선언했습니다.

private enum 으로 선언한 이유는
접근제한자를 습관적으로 붙이면 좋다는 얘길 들어서요.

현 코드에서 private 없어도 동작원리는 같겠지만, 붙여도 문제 없다면
이건 내부에서만 사용된다 라는 꼬리표를 달아준 격인거라 생각합니다.

hanjaesueng replied:
네 이해했습니다 . static으로 선언하지 않으니까 enum에서 변수를 생성할 수가 없네요

wh5929 replied:
> 네 이해했습니다 . static으로 선언하지 않으니까 enum에서 변수를 생성할 수가 없네요

### 답변

네 static 으로 선언한 이유는 아래 링크를 참조하시면 됩니다.
[Swift namespace - struct vs enum](https://sujinnaljin.medium.com/namespace-%EC%A7%80%EC%A0%95%ED%95%98%EA%B8%B0-struct-vs-enum-ce86d5717d95)

wh5929 replied:
> > 네 이해했습니다 . static으로 선언하지 않으니까 enum에서 변수를 생성할 수가 없네요
> 
> ### 답변
> 네 static 으로 선언한 이유는 아래 링크를 참조하시면 됩니다. [Swift namespace - struct vs enum](https://sujinnaljin.medium.com/namespace-%EC%A7%80%EC%A0%95%ED%95%98%EA%B8%B0-struct-vs-enum-ce86d5717d95)

wh5929 replied:
> > > 네 이해했습니다 . static으로 선언하지 않으니까 enum에서 변수를 생성할 수가 없네요
> > 
> > 
> > ### 답변
> > 네 static 으로 선언한 이유는 아래 링크를 참조하시면 됩니다.
[Swift namespace - struct vs enum](https://sujinnaljin.medium.com/namespace-%EC%A7%80%EC%A0%95%ED%95%98%EA%B8%B0-struct-vs-enum-ce86d5717d95)

hanjaesueng commented on GatheRunner/Presentaion/Running/ViewModel/RunGuideViewModel.swift:
```diff
@@ -0,0 +1,58 @@
+
+import SwiftUI
+
+class RunGuideViewModel: ObservableObject {
+
+    // MARK: Internal
+
+    var getArrsRunGuideExperienceItem: [RunGuideItem] {
+        runGuideExperienceItem
+    }
+
```
질문 
----------------
이부분에 ObservableObject를 당장에 채택할 이유는 없어보입니다. @Published가 없어서 나중에 추가할 계획이여서 ObservableObject를 채택하신건가요??

wh5929 committed:
### 코드 수정

1. RunGuideViewModel 클래스에서 ObservableObject 채택되어 있던 부분 제거
2. Hide ViewModifier 의 inVisible 인자값으로 들어오는 부분이 문맥에 맞도록 변경
예) View().hide(true) 일때 객체가 숨겨지도록 수정
3. MapView hide가 정상적으로 되지 않던 문제 해결 -> RunGuideTabView의 @State var isPresentedRunGuideDetailDescriptionView 변수를 -> @Binding 속성으로 변경하여 JustStartView의 @State var isPresentedRunGuideDetailDescriptionView 변수로부터 정상적으로 바인딩 가능
4.

### 리팩토링

1. RunGuideDetailDescriptionView에서 이벤트 실행 코드가 한줄일 경우 보기 좋게 코드 정리
2. RunGuideDetailDescriptionView의 몇몇 부자연스러운 변수명 수정
3. JustStartView의 isPresented 변수명 수정

wh5929 committed:
### 코드 수정

1. RunGuideDetailDescriptionVIew 의 dismiss 방법 변경
기존 : @Environment(\.dismiss) private var dismiss  활용해서 조건에 따라 dismiss 하던 방식 -> 변경 : RunGuideDetailDescriptionVIew 에서 selectedCardItem 변수를 바인딩 받도록해 조건에 따라 nil값을 넣어주는 방식으로 변경(애플 fullScreenCover 문서 참고)

### 버그 수정

1. fullScreenCover 가 닫힐때 불완전하게 내려오던 증상 해결
-> 커스텀 스크롤을 구현하려 화면을 구성하던 뷰 객체의 offset을 변경하여 위치가 이동되었는데, 닫힐때 그 위치가 반영되어 있는것을 확인
-> closeButton 탭 이벤트를 통해 fullScreenCover를 닫을때 yAxisOffsetOfView = 0 으로 설정하여 정상적으로 트랜지션이 작동하도록 설정함

### 리팩토링

1. RunGuideDetailDesciptionView의 몇몇 리터럴 상수명 변경

wh5929 committed:
### 리팩토링

1. RunGuideDetailDescriptionView 의 바인딩 변수명 변경

wh5929 committed:
### 리팩토링

1. RunningView 변수/함수명 변경
2. viewSelectorButton 구조 변경

wh5929 committed:
### 리팩토링

1. UX 향상 작업MainTab - 러닝 뷰 -> 활동 뷰 등 -> 러닝 뷰로 탭 전환시 기존에 보고 있던 뷰를 볼 수 있도록 구성
-> 환경변수 사용

wh5929 committed:
### 리팩토링

1. 변수/함수명 변경

### UI 변경

1. RunGuideDetailDescriptionView UI 변경
2. CardView UI 변경
3. MapView UI 변경

wh5929 committed:
Merge branch 'develop' into feature/RGB2022101401

wh5929 committed:
Revert "Merge branch 'develop' into feature/RGB2022101401"

This reverts commit 5155d264c66d8eeb0018688c39fea272981cb6a7, reversing
changes made to 09b7c3db3ef4654fd64744b8f69e91e240b9c7b9.
